maintenance/wikia/HttpsMigration/createEnAliasForMigratedWikis.php

```diff
@@ -0,0 +1,134 @@
+		$this->mDescription = 'Creates /en alias domain for migrated English wikis';
+		$this->addArg( 'file', 'CSV file with the list of wikis' );
+		$this->addOption( 'saveChanges', 'Create /en alias domains for real.', false, false, 'd' );
+	}
+
+	public function execute() {
+		global $wgUser;
+
+		$fileName = $this->getArg( 0 );
+		$saveChanges = $this->hasOption( 'saveChanges' );
+
+		$fileHandle = fopen( $fileName, 'r' );
+
+		if ( $fileHandle === false ) {
+			$this->output( "Invalid file provided\n" );
+			exit( 1 );
+		}
+
+		$wgUser = User::newFromName( Wikia::BOT_USER );
+
+		while ( ( $data = fgetcsv( $fileHandle ) ) !== false ) {
+			if ( is_null( $data[0] ) ) {
+				continue;
+			}
+
+			$sourceWikiId = $data[0];
+
+			if ( !WikiFactory::isPublic( $sourceWikiId ) ) {
+				$this->output( "Wiki with ID {$sourceWikiId} was not found or is closed!\n" );
+				continue;
+			}
+
+			$sourceDomain = wfNormalizeHost( parse_url( WikiFactory::cityIDtoDomain( $sourceWikiId ), PHP_URL_HOST ) );
+
+			if ( !$sourceDomain ) {
+				$this->output( "Wiki with ID {$sourceWikiId} was not found!\n" );
+				continue;
+			}
+
+			$additionalTargetDomain = $this->getAdditionalDomain( $sourceDomain );
+			if ( !$additionalTargetDomain ) {
+				continue;
+			}
+
+
+			if ( $saveChanges ) {
+
+				if ( !empty($additionalTargetDomain) ){
+					WikiFactory::addDomain( $sourceWikiId, $additionalTargetDomain, 'Creating /en alias for Wiki with ID {$sourceWikiId}' );
+					$this->purgeCachesForWiki( $sourceWikiId );
+
+				}
+			}
+
+			if ( !empty($additionalTargetDomain)  ){
+				$this->output( "/en alias was created for wiki with ID {$sourceWikiId}: {$additionalTargetDomain}\n" );
+			}
+		}
+
+		fclose( $fileHandle );
+	}
+
+	private function getAdditionalDomain( $sourceDomain ) {
+		global $wgFandomBaseDomain;
+		$parts = explode( '.', $sourceDomain );
+		if ( $parts[1] !== 'fandom' ){
+			$this->output( "Failed to create /en alias for {$sourceDomain}!\n" );
+			return false;
+		}
+		return "{$parts[0]}.{$wgFandomBaseDomain}/en";
+	}
+
+
+	private function purgeCachesForWiki( $wikiId ) {
+		global $wgMemc;
+
+		WikiFactory::clearCache( $wikiId );
+
+		Wikia::purgeSurrogateKey( Wikia::wikiSurrogateKey( $wikiId ) );
+		Wikia::purgeSurrogateKey( Wikia::wikiSurrogateKey( $wikiId ), 'mercury' );
+
+		$dbName = WikiFactory::IDtoDB( $wikiId );
+
+		$keys = [
+			wfSharedMemcKey( 'globaltitlev1', $wikiId ),
+			wfSharedMemcKey( 'globaltitlev1:https', $wikiId ),
+			wfForeignMemcKey(
+				$dbName,
+				false,
+				'Wikia\Search\TopWikiArticles',
+				'WikiaSearch',
+				'topWikiArticles',
+				$wikiId,
+				Wikia\Search\TopWikiArticles::TOP_ARTICLES_CACHE,
+				false
+			),
+			wfForeignMemcKey(
+				$dbName,
+				false,
+				'Wikia\Search\TopWikiArticles',
+				'WikiaSearch',
+				'topWikiArticles',
+				$wikiId,
+				Wikia\Search\TopWikiArticles::TOP_ARTICLES_CACHE,
+				true
+			),
+		];
+
+		foreach ( $keys as $key ) {
+			$wgMemc->delete( $key );
+		}
```

Comment:
Why do we need to purge memcache entries? If I understand the script correctly, we only want to add an "alias" domain for English wikis. `foo.fandom.com/en` links were not (I won't be) generated for a given wiki, as MediaWiki will always use a base domain when generating URLs.